### DISCUSSION TITLE
Type hints

### DISCUSSION THREAD

**Discussion by lunacd** [gitpython-developers/gitdb#108] [Q&A] [answered]
Would this library be interested in type hints being added? Is there a reason GitPython library have all type hints ready but this part?

If people are open to type hints, I would be glad to help add those.

**Comment by Byron** [comment, marked as answer]
Ideally, `gitdb` wouldn't be used anymore, even from GitPython. Apparently that's not the case in practice though so having types that make it easier to work with seems useful nonetheless.

My recommendation is to restrict any contributed types, which are welcome, to the ones ordinary users would encounter while interacting with GitPython and its default object backend (the one that uses `git`), and avoid putting any work in types that are related to `gitdb` being used as pure object database as that shouldn't be done. See #106 as example for how broken it is, besides its sluggish performance.

**Reply by lunacd** [reply]
Thank you for explaining the situation. If the ideal situation is to sunset this, I probably wouldn't try to add any hint to this.
